### PR TITLE
RFC - add metadata to job decorator

### PR DESCRIPTION
**Pull request by jamiedemaria**
poking around with adding metadata to the job decorator and converting text and json metadata to tags in JobDefinition

replaced by #7613 - see that pr instead

### CODE REVIEW

**Comment by prha** [comment]
I think this looks good!

I think next steps are probably to display this metadata on the job root page, kind of in a uniform way.  So that probably means threading the metadata through to the ExternalPipeline and adding fields to query it in on the graphql object.

I'm not so sure about the `searchable` parts of the metadata.  I think we can start off by exposing everything and then see  if we even need to distinguish between searchable and non-searchable metadata.

File: python_modules/dagster/dagster/core/definitions/metadata/__init__.py [hunk 1]
```diff
@@ -169,6 +170,10 @@ def emit_metadata(context, df):
             )
     """
 
+    @property
```
Comment (prha) [inline python_modules/dagster/dagster/core/definitions/metadata/__init__.py:173]: let's remove searchable until we know how it's going to be used

**Review by prha — CHANGES_REQUESTED** [review]
See inline comments

**Comment by jamiedemaria** [comment]
closing in favor of https://github.com/dagster-io/dagster/pull/7849

**Pull request closed without merging**